File: XMLCheck/Handler/CheckRulesXML09/Xml9RuleBase.cs
using XMLCheck.Models;
using XMLCheck.Services;

namespace XMLCheck.Handler.CheckRulesXML09
{
    public abstract class Xml9RuleBase:IXml9Rule
    {
        public abstract string Key { get; }

        public abstract ErrorDetails? Check(Xml9Model model,HoSoContext context);

        protected ErrorDetails Error(string message)
        {
            return new ErrorDetails
            {
                key = Key,
                message = message
            };
        }
    }
}

File: XMLValidator_v2/src/parsers/rules/xml9/xml9RuleBase.ts
import { Xml9Model } from '../../../interfaces/xml-models.interface';
import { HoSoContext } from '../../../services/hoso-context';
import { ErrorDetails } from '../../../dto/xml-check.dto';
import { BaseRule } from '../base.rule';

export interface IXml9Rule {
  readonly key: string;
  readonly isEnabled: boolean;
  check(model: Xml9Model, context: HoSoContext): ErrorDetails | null;
}

export abstract class Xml9RuleBase extends BaseRule<Xml9Model> implements IXml9Rule {}
